common/libs/units.ts

```diff
@@ -1,105 +1,28 @@
-import Big, { BigNumber } from 'bignumber.js';
+import BN from 'bn.js';
 import { Token } from 'config/data';
 
-const Units = {
-  wei: '1',
-  kwei: '1000',
-  ada: '1000',
-  femtoether: '1000',
-  mwei: '1000000',
-  babbage: '1000000',
-  picoether: '1000000',
-  gwei: '1000000000',
-  shannon: '1000000000',
-  nanoether: '1000000000',
-  nano: '1000000000',
-  szabo: '1000000000000',
-  microether: '1000000000000',
-  micro: '1000000000000',
-  finney: '1000000000000000',
-  milliether: '1000000000000000',
-  milli: '1000000000000000',
-  ether: '1000000000000000000',
-  kether: '1000000000000000000000',
-  grand: '1000000000000000000000',
-  einstein: '1000000000000000000000',
-  mether: '1000000000000000000000000',
-  gether: '1000000000000000000000000000',
-  tether: '1000000000000000000000000000000'
-};
-
 export type TUnit = typeof Units;
 export type UnitKey = keyof TUnit;
 
-class Unit {
-  public unit: UnitKey;
-  public amount: BigNumber;
-
-  constructor(amount: BigNumber, unit: UnitKey) {
-    this.unit = unit;
-    this.amount = amount;
-  }
-
-  public toString(base?: number) {
-    return this.amount.toString(base);
-  }
-
-  public toPrecision(precision?: number) {
-    return this.amount.toPrecision(precision);
-  }
-
-  public toWei(): Wei {
-    return new Wei(toWei(this.amount, this.unit));
-  }
+export type Wei = BN;
+export const Wei = (input: number | string | BN, base: number = 10): Wei =>
+  new BN(input, base);
```

Comment:
There will very likely be spots where decimals are coming in here. I think it'll be hard to eyeball, so we should probably pop an alert in here if it's a number or string with a `.` in it, and try to cover as much ground as we can.